File: shinjuku-go-20170413/gen.cs
namespace TyphenApi.Type.Submarine.Battle
{
    [MessagePack.MessagePackObject]
    [Newtonsoft.Json.JsonObject(Newtonsoft.Json.MemberSerialization.OptIn)]
    public partial class Ping : TyphenApi.TypeBase<Ping>, TyphenApi.IRealTimeMessage
    {
        [TyphenApi.QueryStringProperty("message", false)]
        [MessagePack.Key("message")]
        [Newtonsoft.Json.JsonProperty("message")]
        [Newtonsoft.Json.JsonRequired]
        public string Message { get; set; }
    }
}

File: shinjuku-go-20170413/gen.go
type Ping struct {
	Message string `json:"message" msgpack:"message"`
}

func (t *Ping) Coerce() error {
	return nil
}

func (t *Ping) Bytes(serializer typhenapi.Serializer) ([]byte, error) {
	if err := t.Coerce(); err != nil {
		return nil, err
	}
	data, err := serializer.Serialize(t)
	if err != nil {
		return nil, err
	}
	return data, nil
}
